packages/fast-usdc/src/exos/advancer.js

```diff
@@ -173,31 +185,33 @@ export const prepareAdvancerKit = (
       },
       depositHandler: {
         /**
-         * @param {NatAmount} amount amount returned from deposit
-         * @param {{ destination: ChainAddress; payment: Payment<'nat'> }} ctx
+         * @param {undefined} result
+         * @param {{ amount: Amount<'nat'>; destination: ChainAddress; tmpSeat: ZCFSeat }} ctx
          */
-        onFulfilled(amount, { destination }) {
+        onFulfilled(result, { amount, destination }) {
           const { poolAccount } = this.state;
-          const transferV = E(poolAccount).transfer(
-            destination,
-            /** @type {DenomAmount} */ ({
-              denom: usdc.denom,
-              value: amount.value,
-            }),
-          );
+          const transferV = E(poolAccount).transfer(destination, {
+            denom: usdc.denom,
+            value: amount.value,
+          });
           return watch(transferV, this.facets.transferHandler, {
             destination,
             amount,
           });
         },
         /**
          * @param {Error} error
-         * @param {{ destination: ChainAddress; payment: Payment<'nat'> }} ctx
+         * @param {{ amount: Amount<'nat'>; destination: ChainAddress; tmpSeat: ZCFSeat }} ctx
          */
-        onRejected(error, { payment }) {
-          // TODO return live payment from ctx to LP
+        onRejected(error, { tmpSeat }) {
+          // TODO return seat allocation from ctx to LP?
           log('🚨 advance deposit failed', q(error).toString());
-          log('TODO live payment to return to LP', q(payment).toString());
+          // TODO #10510 (comprehensive error testing) determine
```

Comment:
this stuff is why #10510 looks to me like "the other 80% of the work".

The local deposit would only fail due to things in our control at design-time (i.e. bugs), but the remote transfer can fail for runtime reasons (timeout).